web/src/main/java/it/pagopa/selfcare/onboarding/web/controller/TokenV2Controller.java

```diff
@@ -157,4 +161,21 @@ public ResponseEntity<Void> delete(@ApiParam("${swagger.tokens.tokenId}")
         tokenService.rejectOnboarding(onboardingId, null);
         return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
     }
+
+
+    @GetMapping(value = "/{onboardingId}/contract", produces = MediaType.APPLICATION_OCTET_STREAM_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = "", notes = "${swagger.tokens.getContract}")
+    public ResponseEntity<byte[]> getContract(@ApiParam("${swagger.tokens.onboardingId}")
+                                              @PathVariable("onboardingId")
+                                              String onboardingId) throws IOException {
+        log.trace("getContract start");
+        log.debug("getContract onboardingId = {}", onboardingId);
+        Resource contract = tokenService.getContract(onboardingId);
+        byte[] bytes = contract.getInputStream().readAllBytes();
```

Comment:
you could return a Resource object instead byte....otherwise you should handle the closure of the input stream